### ISSUE TITLE
Sourcelink TFS Git on a HTTP hosted server, authentication issue

### ISSUE THREAD

**Issue by WdeBruin** [dotnet/sourcelink#175]
I can't find it anywhere in the docs, but I'm running into a problem with sourcelink for TFS Git. When stepping into a package, and clicking the download source button, I get a 401.

This is the error:
ERROR: The request failed with code 401 : "Unauthorized".
The file was not found with Source Link.

In fiddler I see a 401 error 
TF400813: Resource not available for anonymous access. Client authentication required. - Microsoft Team Foundation Server

When I copy the URL that it tries to request and paste it in browser, I get the sourcecode that I try to F11 into. Probably because I'm logged in the TFS environment in the browser.

Info:
- HTTP hosted TFS server. Might be the problem? Can't find that in the documentation.
- Also fails when checkbox for Git Credentials Manager is ticked.

Do I miss something or is it just that it doesn't work on http hosted tfs instances?
Visual studio is already connected to the repository, can the debugger not use a token/cookie from that login?

**Comment by ctaggart** [comment]
@WdeBruin What version of Visual Studio? What version of Team Foundation Server? Are you able to debug with an HTTP proxy like Fiddler?

**Comment by WdeBruin** [comment]
TFS2018
Visual Studio 2017 15.8.5
Am able to debug with Fiddler, I can see that no authentication headers are added to the http request. It gives a 401 and a error page in the response with title "TF400813: Resource not available for anonymous access. Client authentication required. - Microsoft Team Foundation Server".

Should it work on a HTTP hosted TFS instance?

**Comment by ctaggart** [comment]
Via Fiddler, I can see that an `Authorization: Basic` header is added to the request for me:

![image](https://user-images.githubusercontent.com/80104/46033123-37579b00-c0b2-11e8-9ed4-b4fe720f808c.png)

It is added by Git Credential Manager for Windows. What versions do you have of Git and GCMW? Here is what I'm running:
``` ps1
git --version
git version 2.17.0.windows.1
git credential-manager version
Git Credential Manager for Windows version 1.15.2
```

Probably try updating GCMW. The release notes for 1.17.2 that came out 6 days ago look relevant.
https://github.com/Microsoft/Git-Credential-Manager-for-Windows/releases

It is possible that you may need to add some configuration. Have a look at their docs:
https://github.com/Microsoft/Git-Credential-Manager-for-Windows/blob/master/Docs/CredentialManager.md
https://github.com/Microsoft/Git-Credential-Manager-for-Windows/blob/master/Docs/Configuration.md

**Comment by codeworxOS** [comment]
@ctaggart: Yes, but your request is for a GitHub repository. He is talking about on premise TFS. I have the exakt same problem here. the question is, will visual studio allow sending Git Credential Manager authentication headers for http URLs. Because it looks like the GCM fallback is not triggered for our requests.

**Comment by ctaggart** [comment]
@codeworxOS I'm using an on premise GitHub Enterprise. Microsoft.VisualStudio.Debugger.Utilities.SourceLink.SourceLinkClient will *not* send authentication headers for `http` URLs. Make sure you are using secure `https` URLs.

**Comment by codeworxOS** [comment]
Well, in that case… that's the problem here.
I'm not quite sure how to tell the Microsoft.SourceLink.Tfs.Git package zu embed the https url of the on premise tfs instead of the default http one, but at least now we know, that it never was intended to work on a none ssl connection.

**Comment by WdeBruin** [comment]
Just an update on how this worked out:
I arranged a https binding for the TFS server. After that it worked flawlessly. Only thing was, that I got an error that SourceLinkUrl could not be empty. 
Fixed by using version from another nuget feed, https://dotnet.myget.org/feed/sourcelink/package/nuget/Microsoft.SourceLink.Tfs.Git

As indicated in this issue: https://github.com/dotnet/sourcelink/issues/71

Ticket can be closed. I'm sure that at some point an updated version will go to regular NuGet repository.
